docs/features/logging.md

```diff
@@ -32,9 +42,63 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
 }
 ```
 
-If you have multiple middleware components configured, it is advised to put the `ExceptionHandlingMiddleware` as the first one.  By doing so, unhandled exceptions that might occur in other middleware components will also be logged by the `ExceptionHandlingMiddleware` component.
+If you have multiple middleware components configured, it is advised to put the `ExceptionHandlingMiddleware` as the first one.
+By doing so, unhandled exceptions that might occur in other middleware components will also be logged by the `ExceptionHandlingMiddleware` component.
+
+## Logging incoming requests
+
+The `RequestTrackingMiddleware` class can be added to the <span>ASP.NET</span> Core pipeline to log all received HTTP requests.
+The requests by this middleware component are logged through the `ILogger` implementations that are configured inside the project.
+
+The HTTP request headers and request body are by default logged.
+
+**Example**
+
+`HTTP Request POST http://localhost:5000/weatherforecast completed with 200 in 00:00:00.0191554 at 03/23/2020 10:12:55 +00:00 - (Context: [Content-Type, application/json], [Body, {"today":"cloudy"}])`
+
+**Usage**
+
+To use this middleware, add the following line of code in the `Startup.Configure` method:
+
+```csharp
+public void Configure(IApplicationBuilder app, IWebHostEvironment env)
+{
+    app.UseMiddleware<Arcus.WebApi.Logging.RequestTrackingMiddleware>();
+
+    ...
+    app.UseMvc();
+}
+```
+
+**Configuration**
 
-To get the exceptions logged in Azure Application Insights, configure logging like this when building the `IWebHost` in `Program.cs`:
+Optionally, one can inherit from this middleware component and override the default request header and body extraction to discared one or both.
```

Comment:
This should only be done for sanitization imo or if you prefer an action for sanitization that's also an option

Reply:
You mean that the user should get the opportiunity to sanitize the body? That can be done with overriden method.

Reply:
Sure, but what do you mean with this?
> override the default request header

Reply:
That should be `override the default request headers`.

Reply:
I'm still not sure what we mean with this? 🤔

Reply:
I think that the meaning is that you can inherit from this middleware, add your own functionality which allows you to remove one or more headers that are in the incoming request (or overwrite them).  (But why would you do that ? )

Reply:
For security reasons you'd want to have the option to truncate headers for sure.

Oh now I get it